server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java

```diff
@@ -121,20 +123,30 @@ private static void compactAlignedSeries(
       throws IOException, MetadataException {
     MultiTsFileDeviceIterator.AlignedMeasurmentIterator alignedMeasurmentIterator =
         deviceIterator.iterateAlignedSeries(device);
-    List<String> allMeasurments = alignedMeasurmentIterator.getAllMeasurements();
+    List<String> allMeasurements = alignedMeasurmentIterator.getAllMeasurements();
     List<IMeasurementSchema> measurementSchemas = new ArrayList<>();
-    for (String measurement : allMeasurments) {
+    for (String measurement : allMeasurements) {
       // TODO: use IDTable
-      measurementSchemas.add(
-          IoTDB.metaManager.getSeriesSchema(new PartialPath(device, measurement)));
+      try {
+        measurementSchemas.add(
```

Comment:
What if the series get deleted when compacting aligned series in inner space?

Reply:
IMO, we can category the timing of deleting timeseries and compaction:
1. If the timeseries deletion happens before compaction, all the closed tsfiles will skip those deleted timeseries data and the merged one will not contain them.
2. If the timeseries deletion happens during compaction, as it has already checked the measurements validation before compaction, the deleted timeseries data is still merged in this compaction work. Although it's a waste of the resource, but no program or data error happens. And in next compaction task, the former deleted data will be processed as condition 1.